### PR TITLE
Fixed bug for polytool.poly( ) 

### LINKED ISSUE #12400: polytool.poly() can't raise polynomial to negative power?
The code
```
import sympy
print(sympy.polys.polytools.poly(1 / (1 + sympy.sqrt(2)), sympy.Dummy('x')))
```
should not result in the following, right?
```
Traceback (most recent call last):
  File "Temp.py", line 2, in <module>
    print(sympy.polys.polytools.poly(1 / (1 + sympy.sqrt(2)), sympy.Dummy('x')))
  File "sympy\polys\polytools.py", line 6855, in poly
    return _poly(expr, opt)
  File "sympy\polys\polytools.py", line 6805, in _poly
    _poly(factor.base, opt).pow(factor.exp))
  File "sympy\polys\polytools.py", line 1404, in pow
    result = f.rep.pow(n)
  File "sympy\polys\polyclasses.py", line 446, in pow
    return f.per(dmp_pow(f.rep, n, f.lev, f.dom))
  File "sympy\polys\densearith.py", line 978, in dmp_pow
    return dup_pow(f, n, K)
  File "sympy\polys\densearith.py", line 943, in dup_pow
    raise ValueError("can't raise polynomial to a negative power")
ValueError: can't raise polynomial to a negative power
```

### PR DESCRIPTION
**Pull request by ashishkg0022**
It raised an unwanted error `can't raise polynomial to a negative power` , which was a bug.
Fixes #12400

### CODE REVIEW

**Comment by ashishkg0022** [comment]
@jksuom do this fix the issue ?

**Comment by jksuom** [comment]
I think it is the proper fix. Perhaps you could also split the long line differently.

File: sympy/polys/polytools.py [hunk 1]
```diff
@@ -7023,7 +7023,8 @@ def _poly(expr, opt):
             for factor in Mul.make_args(term):
                 if factor.is_Add:
                     poly_factors.append(_poly(factor, opt))
-                elif factor.is_Pow and factor.base.is_Add and factor.exp.is_Integer:
+                elif factor.is_Pow and factor.base.is_Add and factor.exp.is_Integer \
```
Comment (jksuom) [inline sympy/polys/polytools.py]: The last part of this could be moved to the next line.

**Comment by ashishkg0022** [comment]
ping @jksuom

**Comment by jksuom** [comment]
Thanks! This looks good.

**Pull request merged**